packages/jellyfish-api-core/__tests__/category/icxorderbook/makeOffer.test.ts

```diff
@@ -0,0 +1,467 @@
+      amountFrom: new BigNumber(15),
+      orderPrice: new BigNumber(0.01)
+    }
+    let result: ICXGenericResult = await client.icxorderbook.createOrder(order, [])
+    const createOrderTxId = result.txid
+    await container.generate(1)
+
+    // list ICX orders
+    let orders: Record<string, ICXOrderInfo | ICXOfferInfo> = await client.call('icx_listorders', [], 'bignumber')
+    expect((orders as Record<string, ICXOrderInfo>)[createOrderTxId]).toStrictEqual(
+      {
+        status: ICXOrderStatus.OPEN,
+        type: ICXOrderType.INTERNAL,
+        tokenFrom: symbolDFI,
+        chainTo: order.chainTo,
+        receivePubkey: order.receivePubkey,
+        ownerAddress: order.ownerAddress,
+        amountFrom: order.amountFrom,
+        amountToFill: order.amountFrom,
+        orderPrice: order.orderPrice,
+        amountToFillInToAsset: order.amountFrom.multipliedBy(order.orderPrice),
+        height: expect.any(BigNumber),
+        expireHeight: expect.any(BigNumber)
+      }
+    )
+
+    const accountBTCBeforeOffer = await container.call('getaccount', [accountBTC, {}, true])
+    // make offer to partial amount 10 DFI - taker
+    const offer: ICXOffer = {
+      orderTx: createOrderTxId,
+      amount: new BigNumber(0.1), // 10 DFI = 0.1 BTC
+      ownerAddress: accountBTC
+    }
+
+    result = await client.icxorderbook.makeOffer(offer, [])
+    const makeOfferTxId = result.txid
+    await container.generate(1)
+    const accountBTCAfterOffer = await container.call('getaccount', [accountBTC, {}, true])
+
+    // check fee of 0.01 DFI has been reduced from the accountBTCBefore[idDFI]
+    // Fee = takerFeePerBTC(inBTC) * amount(inBTC) * DEX DFI per BTC rate
+    expect(accountBTCAfterOffer[idDFI]).toStrictEqual(accountBTCBeforeOffer[idDFI] - 0.01)
+
+    // List the ICX offers for orderTx = createOrderTxId and check
+    orders = await client.call('icx_listorders', [{ orderTx: createOrderTxId }], 'bignumber')
+    expect(Object.keys(orders).length).toBe(2) // extra entry for the warning text returned by the RPC atm.
+    expect((orders as Record<string, ICXOfferInfo>)[makeOfferTxId]).toStrictEqual(
+      {
+        orderTx: createOrderTxId,
+        status: ICXOrderStatus.OPEN,
+        amount: offer.amount,
+        amountInFromAsset: offer.amount.dividedBy(order.orderPrice),
+        ownerAddress: offer.ownerAddress,
+        takerFee: offer.amount.multipliedBy(ICX_TAKERFEE_PER_BTC).multipliedBy(DEX_DFI_PER_BTC_RATE),
+        expireHeight: expect.any(BigNumber)
+      }
+    )
+
+    // check accountDFI[idDFI] balance
+    const accountDFIAfterOffer = await container.call('getaccount', [accountDFI, {}, true])
+    expect(accountDFIAfterOffer[idDFI]).toStrictEqual(accountDFIBeforeOrder[idDFI] - 15)
+  })
+
+  it('should make an partial offer to an sell BTC order', async () => {
+    const accountDFIBeforeOrder = await container.call('getaccount', [accountDFI, {}, true])
+    // create offer - maker
+    const order: ICXOrder = {
```

Comment:
```suggestion
    const createOrderMaker: ICXOrder = {
```

You should put that code comment into the variable

Reply:
the reason I put those comments is that to give a kind of separation for each sub-section within the test. If i remove the comment and add it only to the variable, that clarity/information is gone or not represented strong enough IMO.
example 
```
it('should make an partial offer to an sell BTC order', async () => {
    const accountDFIBeforeOrder = await container.call('getaccount', [accountDFI, {}, true])
    // create offer - maker
    const order: ICXOrder = {
      chainFrom: 'BTC',
      tokenTo: idDFI,
      ownerAddress: accountDFI,
      amountFrom: new BigNumber(2),
      orderPrice: new BigNumber(100)
    }
    let result: ICXGenericResult = await client.icxorderbook.createOrder(order, [])
    const createOrderTxId = result.txid
    await container.generate(1)

    // list ICX orders
    let orders: Record<string, ICXOrderInfo | ICXOfferInfo> = await client.call('icx_listorders', [], 'bignumber')
    expect((orders as Record<string, ICXOrderInfo>)[createOrderTxId]).toStrictEqual(
      {
        status: ICXOrderStatus.OPEN,
        type: ICXOrderType.EXTERNAL,
        tokenTo: symbolDFI,
        chainFrom: order.chainFrom,
        ownerAddress: order.ownerAddress,
        amountFrom: order.amountFrom,
        amountToFill: order.amountFrom,
        orderPrice: order.orderPrice,
        amountToFillInToAsset: order.amountFrom.multipliedBy(order.orderPrice),
        height: expect.any(BigNumber),
        expireHeight: expect.any(BigNumber)
      }
    )
```

VS

```
it('should make an partial offer to an sell BTC order', async () => {
    const accountDFIBeforeOrder = await container.call('getaccount', [accountDFI, {}, true])
    const createOrderMaker: ICXOrder = {
      chainFrom: 'BTC',
      tokenTo: idDFI,
      ownerAddress: accountDFI,
      amountFrom: new BigNumber(2),
      orderPrice: new BigNumber(100)
    }
    let result: ICXGenericResult = await client.icxorderbook.createOrder(createOrderMaker, [])
    const createOrderTxId = result.txid
    await container.generate(1)

    let retrivedOrders: Record<string, ICXOrderInfo | ICXOfferInfo> = await client.call('icx_listorders', [], 'bignumber')
    expect((retrivedOrders as Record<string, ICXOrderInfo>)[createOrderTxId]).toStrictEqual(
      {
        status: ICXOrderStatus.OPEN,
        type: ICXOrderType.EXTERNAL,
        tokenTo: symbolDFI,
        chainFrom: order.chainFrom,
        ownerAddress: order.ownerAddress,
        amountFrom: order.amountFrom,
        amountToFill: order.amountFrom,
        orderPrice: order.orderPrice,
        amountToFillInToAsset: order.amountFrom.multipliedBy(order.orderPrice),
        height: expect.any(BigNumber),
        expireHeight: expect.any(BigNumber)
      }
    )
```

Reply:
> the reason I put those comments is that to give a kind of separation for each sub-section within the test. If i remove the comment and add it only to the variable, that clarity/information is gone or not represented strong enough IMO.
> example
> 
> ```
> it('should make an partial offer to an sell BTC order', async () => {
>     const accountDFIBeforeOrder = await container.call('getaccount', [accountDFI, {}, true])
>     // create offer - maker
>     const order: ICXOrder = {
>       chainFrom: 'BTC',
>       tokenTo: idDFI,
>       ownerAddress: accountDFI,
>       amountFrom: new BigNumber(2),
>       orderPrice: new BigNumber(100)
>     }
>     let result: ICXGenericResult = await client.icxorderbook.createOrder(order, [])
>     const createOrderTxId = result.txid
>     await container.generate(1)
> 
>     // list ICX orders
>     let orders: Record<string, ICXOrderInfo | ICXOfferInfo> = await client.call('icx_listorders', [], 'bignumber')
>     expect((orders as Record<string, ICXOrderInfo>)[createOrderTxId]).toStrictEqual(
>       {
>         status: ICXOrderStatus.OPEN,
>         type: ICXOrderType.EXTERNAL,
>         tokenTo: symbolDFI,
>         chainFrom: order.chainFrom,
>         ownerAddress: order.ownerAddress,
>         amountFrom: order.amountFrom,
>         amountToFill: order.amountFrom,
>         orderPrice: order.orderPrice,
>         amountToFillInToAsset: order.amountFrom.multipliedBy(order.orderPrice),
>         height: expect.any(BigNumber),
>         expireHeight: expect.any(BigNumber)
>       }
>     )
> ```
> 
> VS
> 
> ```
> it('should make an partial offer to an sell BTC order', async () => {
>     const accountDFIBeforeOrder = await container.call('getaccount', [accountDFI, {}, true])
>     const createOrderMaker: ICXOrder = {
>       chainFrom: 'BTC',
>       tokenTo: idDFI,
>       ownerAddress: accountDFI,
>       amountFrom: new BigNumber(2),
>       orderPrice: new BigNumber(100)
>     }
>     let result: ICXGenericResult = await client.icxorderbook.createOrder(createOrderMaker, [])
>     const createOrderTxId = result.txid
>     await container.generate(1)
> 
>     let retrivedOrders: Record<string, ICXOrderInfo | ICXOfferInfo> = await client.call('icx_listorders', [], 'bignumber')
>     expect((retrivedOrders as Record<string, ICXOrderInfo>)[createOrderTxId]).toStrictEqual(
>       {
>         status: ICXOrderStatus.OPEN,
>         type: ICXOrderType.EXTERNAL,
>         tokenTo: symbolDFI,
>         chainFrom: order.chainFrom,
>         ownerAddress: order.ownerAddress,
>         amountFrom: order.amountFrom,
>         amountToFill: order.amountFrom,
>         orderPrice: order.orderPrice,
>         amountToFillInToAsset: order.amountFrom.multipliedBy(order.orderPrice),
>         height: expect.any(BigNumber),
>         expireHeight: expect.any(BigNumber)
>       }
>     )
> ```

make sense, thanks!